core/src/test/kotlin/ch/ergon/dope/buildTest/TypeFunctionsTest.kt

```diff
@@ -0,0 +1,154 @@
+package ch.ergon.dope.buildTest
+
+import ch.ergon.dope.QueryBuilder
+import ch.ergon.dope.helper.ParameterDependentTest
+import ch.ergon.dope.helper.someBooleanField
+import ch.ergon.dope.helper.someBucket
+import ch.ergon.dope.helper.someNumber
+import ch.ergon.dope.helper.someNumberArrayField
+import ch.ergon.dope.helper.someNumberField
+import ch.ergon.dope.helper.someStringField
+import ch.ergon.dope.resolvable.expression.unaliased.type.access.get
+import ch.ergon.dope.resolvable.expression.unaliased.type.arithmetic.add
+import ch.ergon.dope.resolvable.expression.unaliased.type.toDopeType
+import ch.ergon.dope.resolvable.expression.unaliased.type.typefunction.isArray
+import ch.ergon.dope.resolvable.expression.unaliased.type.typefunction.isAtom
+import ch.ergon.dope.resolvable.expression.unaliased.type.typefunction.isBoolean
+import ch.ergon.dope.resolvable.expression.unaliased.type.typefunction.isNumber
+import ch.ergon.dope.resolvable.expression.unaliased.type.typefunction.isString
+import ch.ergon.dope.resolvable.expression.unaliased.type.typefunction.toArray
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class TypeFunctionsTest : ParameterDependentTest {
+    private lateinit var create: QueryBuilder
+
+    @BeforeTest
+    fun setup() {
+        create = QueryBuilder()
+    }
+
+    @Test
+    fun `should support ISARRAY as boolean comparison`() {
+        val expected = "SELECT * FROM `someBucket` WHERE ISARRAY(`numberArrayField`)"
+
+        val actual = create
+            .selectFrom(
+                someBucket(),
+            )
+            .where(
+                isArray(someNumberArrayField()),
```

Comment:
I would go for the syntax
```
...where(someNumberArrayField().isArray())
...where(someNumberArrayField().isAtom().and(someNumberArrayField().isBoolean())
...where(someNumberArrayField().toBoolean().isNumber())
```

Reply:
I added this syntax for all `is...` and `to...` extensions. I left `typeOf(...)` as is for a more intuitive usage.